### PR TITLE
refactor: replace GameScenario with a more generic Game type

### PR DESCRIPTION
**Pull request by Muthaias**
This branch should clean up some component code and prepare for use of a more generic Game concept. This PR will:
* Remove the GameScenario concept
* Add the Game\<P\> concept
* Make BasicGameScenario use the Game\<P\> concept
* Make Game component use Game\<P\> concept
* Remove type specificity from components Deck and Card

This PR should not change any user facing behaviour.

To test:
* Build and run `npm start`
* Fixed by new content build. ~Make sure you are using latest data with `modifiers` and not `modifier` (currently the target online data is a mismatch)~
* Verify that the game behaves as expected

### CODE REVIEW

File: src/game/GameScenario.ts [hunk 1]
```diff
@@ -324,12 +324,47 @@ export class BasicGameScenario implements GameScenario {
...
+                    modifier: (state) => {
+                        return this.getUpdatedState(
+                            state,
+                            data,
+                            data.actions.left,
+                        )
+                    },
+                },
+                right: {
+                    description: data.actions.right.description ?? 'No',
```
Comment (Greenheart) [inline src/game/GameScenario.ts:349]: This default value should be "Yes"

File: src/game/Types.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        left: {
+            description: string
+        }
+        right: {
+            description: string
+        }
+    }
+}
+
+export interface Card<P> extends CardPresentation {
```
Comment (Greenheart) [inline src/game/Types.ts:31]: This separation between presentation data and runtime logic is really nice!

File: src/game/GameScenario.ts [hunk 3]
```diff
@@ -51,26 +43,35 @@ export class BasicGameScenario implements GameScenario {
...
+        return this._scenario.stats.map<Stat<WorldState>>((stat) => ({
+            ...stat,
+            getValue: ({ params }) => params.state[stat.id],
+        }))
     }
 
     /**
      * Get the initial state for a given scenario.
      *
      * @param scenario The scenario default data which holds all cards, events and similar
```
Comment (Greenheart) [inline src/game/GameScenario.ts:56]: Just noticed we have outdated docs here. Could you remove the `@param` line?

File: src/game/GameScenario.ts [hunk 4]
```diff
@@ -84,19 +85,18 @@ export class BasicGameScenario implements GameScenario {
      * @param action The player's choosen action for how to move forward in the game
      */
     getUpdatedState(
-        prevState: GameState,
```
Comment (Greenheart) [inline src/game/GameScenario.ts:87]: Outdated docs here too, Remove the `@param scenario` line

File: src/game/GameScenario.ts (hunk 1 above)
Comment (Greenheart) [reply src/game/GameScenario.ts:349]: Also interesting how this data seems to override `CardView.tsx`. Maybe this is supposed to be changed in the next PR #157 ?

**Review by Greenheart — CHANGES_REQUESTED** [review]
After experimenting a bit it's clear that this will make it much easier to simplify the game runtime from the content utils in the future. With TS we can ensure the shared data layer works, while also having the freedom to create relevant abstractions for both workflows.

Great work on this - thanks for taking the initiative! 🌱✨

TODO:
- Resolve comments about default card action description value.
- Remove two outdated parameters from docs.

Apart from that, everything seems to work well from what I can tell right now!

File: src/game/GameScenario.ts (hunk 4 above)
Comment (Muthaias) [reply src/game/GameScenario.ts:87]: This documentation var already out of date before these changes. This should be fixed in another PR

File: src/game/GameScenario.ts (hunk 3 above)
Comment (Muthaias) [reply src/game/GameScenario.ts:56]: This is a previous mistake not related to this PR. I'd rather fix it separately

Comment (Greenheart) [reply src/game/GameScenario.ts:56]: Fair enough :)

**Comment by Greenheart** [comment]
I'd say this is ready to merge. Anything local unpublished changes to add before merging?

**Review by Greenheart — APPROVED** [review]

**Pull request merged**